aspnetcore/blazor/call-web-api.md

```diff
@@ -0,0 +1,230 @@
+  * `FetchCredentialsOption.SameOrigin` ("same-origin") &ndash; Advises the browser to send credentials (such as cookies or HTTP auth headers) only if the target URL is on the same origin as the calling application.
+
+```cshtml
+@using System.Net.Http.Headers
+@using Microsoft.AspNetCore.Blazor.Http
+@inject HttpClient Http
+
+@functions {
+    private async Task PostRequest()
+    {
+        Http.DefaultRequestHeaders.Authorization =
+            new AuthenticationHeaderValue("Bearer", "{OAUTH TOKEN}");
+
+        var requestMessage = new HttpRequestMessage()
+        {
+            Method = new HttpMethod("POST"),
+            RequestUri = new Uri("https://localhost:10000/api/todo"),
+            Content = 
+                new StringContent(
+                    @"{""name"":""A New Todo Item"",""isComplete"":false}")
+        };
+
+        requestMessage.Content.Headers.ContentType = 
+            new System.Net.Http.Headers.MediaTypeHeaderValue(
+                "application/json");
+
+        requestMessage.Content.Headers.TryAddWithoutValidation(
+            "x-custom-header", "value");
+        
+        requestMessage.Properties[WebAssemblyHttpMessageHandler.FetchArgs] = new
+        { 
+            credentials = FetchCredentialsOption.Include
+        };
+
+        var response = await Http.SendAsync(requestMessage);
+        var responseStatusCode = response.StatusCode;
+        var responseBody = await response.Content.ReadAsStringAsync();
+    }
+}
+```
+
+For more information on Fetch API options, see [MDN web docs: Window​OrWorker​Global​Scope​.fetch():Parameters](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#Parameters).
+
+When sending credentials (authorization cookies/headers) on CORS requests, allow the `Authorization` header when creating the CORS policy for CORS middleware. The following policy includes configuration for:
+
+* Request origins (`http://localhost:5000`, `https://localhost:5001`).
+* Any method (verb).
+* `Content-Type` and `Authorization` headers. To allow a custom header (for example, `x-custom-header`), list the header when calling <xref:Microsoft.AspNetCore.Cors.Infrastructure.CorsPolicyBuilder.WithHeaders*>.
+* Credentials set by client-side JavaScript code (`credentials` property set to `include`).
+
+```csharp
+app.UseCors(policy => 
+    policy.WithOrigins("http://localhost:5000", "https://localhost:5001")
+    .AllowAnyMethod()
+    .WithHeaders(HeaderNames.ContentType, HeaderNames.Authorization, "x-custom-header")
+    .AllowCredentials());
+```
+
+For more information, see <xref:security/cors> and the sample app's HTTP Request Tester component (*Components/HTTPRequestTester.razor*).
+
+## Cross-origin resource sharing (CORS)
+
+For more information on making cross-origin resource sharing (CORS) requests, see <xref:security/cors>. The sample app demonstrates CORS. See the Call Web API component (*Pages/CallWebAPI.razor*).
+
+## HTTP Request Tester
```

Comment:
I think we can remove this section. We already point folks to the sample app at the beginning.